transformers4rec/torch/model/base.py

```diff
@@ -524,6 +521,17 @@ def forward(
         self, inputs: TensorOrTabularData, targets=None, training=False, testing=False, **kwargs
     ):
         # TODO: Optimize this
+        # Convert inputs to float32 which is the default type, expected by PyTorch
+        for name, val in inputs.items():
```

Comment:
By default, PyTorch set the model's parameters as scalars (`float32`) and expects the input data to be of  type`float32`. 
If the inputs are Double (i.e. float64), the user should explicitly set the model to use Double parameters using `model.double()`. 
I am not sure if there is a use-case/benefit of using `Double` instead of scalar inputs? with the old default type convention, we were always using scalar inputs. That's why I added this conversion within the Model's forward method to always set the inputs as scalars.